sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -400,6 +415,31 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     // metadata.
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
+  // Add optimization flags.
+  const char *optLevelStr = getUint32PropAsOptStr(Img, "optLevel").c_str();
+  // TODO: Passing these options to vector compiler causes build failure in
+  // backend. Will pass the flags once backend compilation issue is resolved.
+  // Update only if compile options are not overwritten by environment
+  // variable.
+  if (!isEsimdImage && !CompileOptsEnv && optLevelStr != nullptr &&
+      optLevelStr[0] != '\0') {
+    // Making sure all devices have the same platform.
+    assert(!Devs.empty() &&
+           std::all_of(Devs.begin(), Devs.end(), [&](const device &Dev) {
+             return Dev.get_platform() == Devs[0].get_platform();
+           }));
+    const char *backend_option = nullptr;
+    // Empty string is returned in backend_option when no appropriate backend
+    // option is available for a given frontend option.
+    Plugin.getBackendOption(
+        detail::getSyclObjImpl(Devs[0].get_platform())->getHandleRef(),
+        optLevelStr, &backend_option);
```

Comment:
Please check error in return

Reply:
This function does not return error back to caller. I have checked for error inside the function. Hope that's agreeable.

Thanks